Sources/Workspace/PinsStore.swift

```diff
@@ -61,27 +81,45 @@ public struct PinsStore {
     public var pins: AnySequence<Pin> {
         return AnySequence<Pin>(pinsMap.values)
     }
+    
+    /// Most recent error, if any.
+    /// FIXME: This needs to be an array of diagnostics, or perhaps a structured
+    /// log of what-all went on while trying to instantiate the PinStore.
+    public var error: Error?
+    
+    /// Returns true if and only iff there are any errors; when this property is
+    /// true, mutating PinStore operations are disallowed and will throw errors.
+    /// FIXME: We should have a protocol for "things that can hold errors", and
+    /// then the `hasErrors` property should be a convenience implementation so
+    /// it doesn't need to be implemented in every adopter of the protocol.
+    public var hasError: Bool { return error != nil }
```

Comment:
It's an interesting idea, and I see your point about making the client really have to deal with errors, but I'm not sure how well this work work in practice.  That seems more cumbersome to me than having a single object that could either be in an error or non-error state, especially since we hope that (through fixits etc) the object could go from being in an error state to no longer being in an error state.  In that case, PinsStore objects would come and go instead of just existing, which could prevent caching of state etc, and also there would need to be some way of updating the client's account of the error.  It also wouldn't work so well when multiple clients had references to the same object; would each of them hold a copy of the error?

Also, although the focus here is on making sure that initializers don't throw, we also want to handle the case of an object that was initially valid go into an error state.

Perhaps I misunderstand what you're suggesting.